lib/Core/StatsTracker.cpp

```diff
@@ -242,13 +254,19 @@ StatsTracker::StatsTracker(Executor &_executor, std::string _objectFilename,
   }
 
   if (OutputStats) {
-    statsFile = executor.interpreterHandler->openOutputFile("run.stats");
-    assert(statsFile && "unable to open statistics trace file");
+    if(sqlite3_open(executor.interpreterHandler->getOutputFilename("run.stats").c_str(), &statsFile)){
+       std::stringstream error_s;
+       error_s << "Can't open database: " << sqlite3_errmsg(statsFile);
+       sqlite3_close(statsFile);
+       klee_error("%s", error_s.str().c_str());
```

Comment:
Why don't you just put the sqlite3_errmsg here for s instead of the stringstream?

Reply:
Wait... otherwise you would use it after _close. That shouldn't work. Keep it that way.

Reply:
You mean, I get the same leak as below? This API is annoying ...

Reply:
I reverted this.